src/main/java/com/uber/cadence/internal/testservice/TestWorkflowMutableStateImpl.java

```diff
@@ -735,8 +742,52 @@ private void fireTimer(String timerId) {
   }
 
   private void processFailWorkflowExecution(
-      RequestContext ctx, FailWorkflowExecutionDecisionAttributes d, long decisionTaskCompletedId)
+      RequestContext ctx,
+      FailWorkflowExecutionDecisionAttributes d,
+      long decisionTaskCompletedId,
+      String identity)
       throws InternalServiceError, BadRequestError {
+    WorkflowData data = workflow.getData();
+    if (data.retryState.isPresent()) {
+      int backoffIntervalSeconds =
+          data.retryState
+              .get()
+              .getBackoffIntervalInSeconds(d.getReason(), store.currentTimeMillis());
+      if (backoffIntervalSeconds > 0) {
+        ContinueAsNewWorkflowExecutionDecisionAttributes continueAsNewAttr =
```

Comment:
Just want to confirm: Retry is implemented as ContinueAsNew only in the test framework, right? In real service it will be still the same run?

Reply:
No, the test framework matches the real service behavior.